client/lib/i18n-utils/switch-locale.js

```diff
@@ -139,6 +140,7 @@ export default function switchLocale( localeSlug ) {
 					}
 
 					i18n.setLocale( body );
+					setLocaleData( body );
```

Comment:
A similar call should be also in the `isDefaultLocale()` branch a few lines above. There, `i18n.configure()` resets the `i18n-calypso` locale to `en`.

Reply:
Is there anything analogous in `@wordpress/i18n`?

Reply:
Would that be analagous to resetting to no translations `setLocaleData()`?

Reply:
Looking at `@wordpress/i18n` [`setLocaleData` source](https://github.com/WordPress/gutenberg/blob/master/packages/i18n/src/index.js#L49-L62), it seems that it doesn't support resetting the library back to the initial state. It can only overwrite existing data, or fill in the defaults, but it never removes anything.

In other words, it seems there's no way to reset the library, say, from `cs` back to `en`. Apparently the existing use cases only ever set the locale once.